### DISCUSSION TITLE
How many insert queries in repeated function call

### DISCUSSION THREAD

**Discussion by elhananjair** [porsager/postgres#754] [Q&A]
Hello friends,
I am still testing postgresjs and it is so easy to work with thanks again @porsager.
The problem I am facing currently is, I am not sure if my insert query is really inserting all data that have been sent to the function I am doing insert with.

I have a function **saveNum** which saves/insert new phone number to database, and just for testing purpose I created counter variable to check if the counter and data from database is equal. But, unfortunately they are not. My question is, does PostgreSQL limits insert queries execute in certain amount of time?  I am curious because I am inserting the data instantly as the process received **external data which sometimes might be 1000 data per second** that arrives from external server.

Therefore should I save such thing into array and run insert query in some time interval? or is there any tip for that? Here is the code I am working with \[Implementing SMPP protocol\]

```
...
let deliveryCounter = 0;
session.on('deliver_sm', (pdu) => {
    if(pdu.esm_class === 4){ 
        let stat = pdu.short_message.message.substring(pdu.short_message.message.search('stat')+5, pdu.short_message.message.search('stat')+12);
        if(stat === 'DELIVRD'){
            console.log(`Delivery report:  ${deliveryCounter++}` + pdu.source_addr);
            saveDelivery(pdu.source_addr).catch(e => console.log(e))
            session.send(pdu.response()); 
        }else {
            saveUndelivered(pdu.source_addr, stat).catch(e => console.log(e))
        }
    }else if(pdu.esm_class !== 4){
        console.log(pdu.short_message);
    }
})

async function saveDelivery(pnum){ 
    try {
        return await sql`
    insert into sms.delivery
      (phone)
    values
      (${pnum})
    returning phone
  `
    } catch (e) {
        console.log("Error Found: " + e.message)
    }

}
...
```

I checked a database and deliveryCounter value and there is some difference.

**Comment by K4leri** [comment]
i have almost the same problem with updating querires
```ts
const result = await sql`
  update activity set ${sql(updateFields, columns)}
  where id = ${id} and uid = ${uid}
`;
```
```ts 
result.count 
Property 'count' does not exist on type 'object'.
```

its very hard to get how many rows were updated. For example in pgAdmin at every operation it says that it insert\update\delete n rows.

The same problem acording delete operation.

**Reply by porsager** [reply]
I don't think this is related to the original post, and the code you posted above is not a reproducible. `.count` is a core feature of Postgres.js, covered in tests, and I also use it extensively myself, so I think you've got an issue somewhere else. If you can post a full reproducible sample I'll take a look ;)

**Comment by porsager** [comment]
Hey @elhananjair ..

Sorry, never saw this post until @K4leri reply, but curious if you found your way through? There is no limit really, and I wouldn't say you should need optimization unless you start to see performance issues. I think you're simply dealing with a race condition between the time it takes for the query to run and the time you update your counter :)

**Reply by elhananjair** [reply]
Hello, @porsager 
I have already finished that project but I couldn't find a solution for that and it didn't work for me. 
I will try it again with a simple code and will share it with you here.
